src/Raven.Server/RavenServer.cs

```diff
@@ -2946,6 +2973,50 @@ public bool CertificateHasWellKnownIssuer(X509Certificate2 cert, out string issu
             return false;
         }
 
+        private bool AreCertificateSansValid(X509Certificate2 cert)
+        {
+            var serverDomain = new Uri(ServerStore.GetNodeHttpServerUrl()).Host;
+            var sans = CertificateUtils.GetCertificateAlternativeNames(cert).ToList();
+            if (sans.Count == 0)
+            {
+                if (_authAuditLog.IsInfoEnabled)
+                {
+                    _authAuditLog.Info("Certificate does not contain any SAN.");
+                }
+
+                return false;
+            }
+
+            Regex regex;
+            foreach (var san in sans)
+            {
+                if (san.StartsWith("*."))
+                {
+                    var array = san.Split("*.");
+                    if (array.Length != 2)
+                    {
+                        if (_authAuditLog.IsInfoEnabled)
+                        {
+                            _authAuditLog.Info($"Certificate {cert.Thumbprint} contains invalid SAN {san}");
+                        }
+
+                        continue;
+                    }
+
+                    regex = new Regex(@"^[^.]+\." + Regex.Escape(array[1]) + "$", RegexOptions.IgnoreCase);
```

Comment:
Replace with `EndsWith`, instead of allocation and using aregex

Reply:
`EndsWith` is not enough, i will have to check if number of subdomains match.
For exampe cert with san `*.test` is not valid for domain `a.b.test`.

Reply:
`serverDomain.EndsWith(array[1]) && serverDomain[0..(serverDomain.len - array[1].len].Contains(".")` - no?